File: Data Structures & Algorithms/two-integer-sum/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) {
        const waitingResults = new Map();
    
        for (let i = 0; i < nums.length; i++) {
            if (waitingResults.has(nums[i])) {
                return [waitingResults.get(nums[i]), i];
            } 
            
            let complemento = target - nums[i];
            waitingResults.set(complemento, i);
        }
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        l, r = 0, len(numbers) - 1
        while l <= r:
            calc = numbers[l] + numbers[r]
            if calc == target:
                return [numbers[l], numbers[r]]
            elif calc > target:
                r -= 1
            else:
                l += 1


        